src/transaction.rs

```diff
@@ -117,16 +113,20 @@ impl<Err> TransactionBuilder<Err> {
         let (result_tx, mut result_rx) = futures_channel::oneshot::channel();
         let (polled_forbidden_thing_tx, mut polled_forbidden_thing_rx) =
```

Comment:
I think you should be able to use a single oneshot channel here since the future can now take a reference on a cell to pass the result (see https://github.com/Ekleog/indexed-db/pull/10/files#diff-ca40038cc1220f9b25a1e0808df841d4b5fafcd936bfd43b9f8e39b8719a66c5R116)

Reply:
I just did it :tada: and also pushed another commit to get rid of most of the other channels, thus unblocking [this opportunity](https://github.com/Ekleog/indexed-db/issues/8#issuecomment-2734818803).

BTW if you have any comments on said opportunity, I'm willing to hear it!